### PR TITLE
Max-width option for figure captions

### LINKED ISSUE #1707: Figure option: Caption length
Sometimes, the fact that the caption of a figure is at the same width as the image can be harmful, as shown in this example:

image

Since the caption is so large it results in a weird-looking page. I'm thinking we add a toggle option that would let a caption either fit to the same width as the image, or, 100% width, as the mockup below is shown (using an H5 instead of the caption):

image

### PR DESCRIPTION
**Pull request by jpeterson976**
Resolves #1707 

Added a select box that lets you choose between image-width and max-width for figure captions. The default is image width, since that's the current behavior.

### CODE REVIEW

**Review by zachberry — CHANGES_REQUESTED** [review]
I didn't review the code but I did test it in the browser. I see it working in the editor but in doesn't seem to work in the viewer. Additionally it doesn't seem to be working for custom sizes (I tried 100x100)

File: packages/obonode/obojobo-chunks-figure/image-properties-modal.test.js [hunk 1]
```diff
@@ -201,7 +201,23 @@ describe('Image Properties Modal', () => {
...
+		const component = mount(
+			<ImageProperties
+				content={{ url: 'mock_url', captionWidth: 'image-width' }}
+				onConfirm={jest.fn}
+			/>
+		)
+
+		// Simulate changing "Caption Width"
+		component
+			.find('#obojobo-draft--chunks--figure--caption-width')
```
Comment (iturgeon) [inline packages/obonode/obojobo-chunks-figure/image-properties-modal.test.js:214]: 👍

File: packages/obonode/obojobo-chunks-figure/__snapshots__/editor-component.test.js.snap [hunk 2]
```diff
@@ -58,14 +58,15 @@ exports[`Figure Editor Node Figure component 1`] = `
...
+exports[`Figure Editor Node Figure component handles tabbing 1`] = `"<div><div class=\\"obojobo-draft--chunks--figure viewer undefined  is-not-selected\\"><div contenteditable=\\"false\\" class=\\"accessibility-warning\\" style=\\"user-select: none;\\">Accessibility Warning: No Alt Text!</div><figure class=\\"container\\"><div class=\\"figure-box   is-not-selected\\" contenteditable=\\"false\\"><div class=\\"obojobo-draft--components--button is-not-dangerous align-center delete-button\\" contenteditable=\\"false\\"><button class=\\"button\\" tabindex=\\"-1\\" contenteditable=\\"false\\">×</button></div><div class=\\"image-toolbar\\"><div class=\\"obojobo-draft--components--button is-not-dangerous align-center properties-button\\" contenteditable=\\"false\\"><button class=\\"button\\" tabindex=\\"-1\\" contenteditable=\\"false\\">Image Properties</button></div></div><div class=\\"img-placeholder\\"></div></div><figcaption class=\\"align-center is-undefined\\"></figcaption></figure></div></div>"`;
```
Comment (iturgeon) [inline packages/obonode/obojobo-chunks-figure/__snapshots__/editor-component.test.js.snap:69]: `is-undefined` seems undesirable here.  Does this indicate we need a default value in the editor component?

Keep in mind that previously created Figure chunks will render `is-undefined` like this sample

**Review by iturgeon — CHANGES_REQUESTED** [review]
It looks like the viewer defaults to image-width for figures missing the caption width setting, the only thing not handling it is the style in the editor that results in `is-undefined` getting rendered to the dom.  We should either default to image-width or just avoid displaying the is-undefined at all

I expanded use of the image-width constants to everywhere they show up in this chunk.

File: packages/obonode/obojobo-chunks-figure/adapter.js [hunk 3]
```diff
@@ -21,6 +23,7 @@ const Adapter = {
 		model.setStateProp('width', null, p => parseInt(p, 10))
 		model.setStateProp('height', null, p => parseInt(p, 10))
 		model.setStateProp('alt', null)
+		model.setStateProp('captionWidth', ImageCaptionWidthTypes.IMAGE_WIDTH)
```
Comment (zachberry) [inline packages/obonode/obojobo-chunks-figure/adapter.js:26]: This should lower-case the attribute and ensure that it only be one of the two acceptable values:

```suggestion
		model.setStateProp('captionWidth', ImageCaptionWidthTypes.IMAGE_WIDTH, p => p.toLowerCase(), [ImageCaptionWidthTypes.IMAGE_WIDTH, ImageCaptionWidthTypes.MAX_WIDTH])
```

File: packages/obonode/obojobo-chunks-figure/viewer-component.js [hunk 4]
```diff
@@ -24,16 +30,18 @@ const Figure = props => {
...
 				className={`obojobo-draft--chunks--figure viewer ${props.model.modelState.size}`}
 			>
 				<div className="container">
-					<figure unselectable="on" style={customStyle}>
-						<Image chunk={props.model} />
+					<figure unselectable="on">
+						<Image chunk={props.model} style={customStyle} />
 						{props.model.modelState.textGroup.first.text.length > 0 ? (
-							<figcaption>
+							<figcaption className={`is-${captionWidth}`} style={captionStyle}>
```
Comment (zachberry) [inline packages/obonode/obojobo-chunks-figure/viewer-component.js:44]: I think we need to change this to `is-caption-width-${captionWidth}`

**Review by zachberry — CHANGES_REQUESTED** [review]
Looks great! I just have a few minor suggestions

**Review by zachberry — APPROVED** [review]

**Pull request merged**